Invalid import path `@/types`

### Describe the issue you're facing

In `@storyblok/react/dist/common/client.d.ts` file there is a import:
```ts
import type { TUseStoryblokState } from '@/types';
```

Such directory does not exist in the package. It could be related to TypeScript cfonfiguration I see in the repository:
```
    "paths": {
      "@/*": ["./src/*"]
    },
```
But there is no `tsconfig.json` in the package. It presents only in GitHub repository https://github.com/storyblok/storyblok-react/blob/main/tsconfig.json

This creates error while running `tsc` with `"skipLibCheck": false,`:
`node_modules/@storyblok/react/dist/common/client.d.ts:1:41 - error TS2307: Cannot find module '@/types' or its corresponding type declarations.`
```
import type { TUseStoryblokState } from '@/types';
```

Also causing `typescript-eslint` to faile with errors like:
```
@typescript-eslint/no-unsafe-assignment
@typescript-eslint/no-unsafe-call
@typescript-eslint/no-unsafe-member-access
```

### Reproduction

https://stackblitz.com/edit/react-sdk-ts-demo-4mvgfj9m?file=package.json

### Steps to reproduce

Run `npm run tsc`

### System Info

```shell
System:
    OS: macOS 15.1.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 2.24 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.12.0 - ~/.nvm/versions/node/v22.12.0/bin/node
    Yarn: 4.5.3 - ~/.nvm/versions/node/v22.12.0/bin/yarn
    npm: 10.9.0 - ~/Developer/home-one/node_modules/.bin/npm
  Browsers:
    Chrome: 131.0.6778.110
    Safari: 18.1.1
```


### Used Package Manager

npm

### Error logs (Optional)

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://www.storyblok.com/trust-center#code-of-conduct?utm_source=github.com&utm_medium=readme&utm_campaign=storyblok-js-client)

--- a/package.json
+++ b/package.json
@@ -34,7 +34,7 @@
   ],
   "scripts": {
     "dev": "vite build --watch",
-    "build": "vite build && tsc --project tsconfig.json",
+    "build": "vite build",
     "test": "pnpm run test:unit:ci && pnpm run test:e2e",
     "test:unit": "vitest",
     "test:unit:ci": "vitest run",
@@ -84,6 +84,7 @@
     "start-server-and-test": "^2.0.8",
     "typescript": "5.6.3",
     "vite": "^5.4.11",
+    "vite-plugin-dts": "^4.3.0",
     "vitest": "^2.1.8"
   },
   "babel": {

--- a/vite.config.ts
+++ b/vite.config.ts
@@ -1,9 +1,10 @@
 import { defineConfig } from 'vitest/config';
 import path from 'node:path';
 import preserveDirectives from 'rollup-plugin-preserve-directives';
+import dts from 'vite-plugin-dts';
 
 export default defineConfig({
-  plugins: [preserveDirectives()],
+  plugins: [dts({ insertTypesEntry: true }), preserveDirectives()],
   build: {
     lib: {
       entry: [

